x-pack/plugins/enterprise_search/common/ml_inference_pipeline/index.test.ts

```diff
@@ -50,3 +60,121 @@ describe('getMlModelTypesForModelConfig lib function', () => {
+    const inferenceType = SUPPORTED_PYTORCH_TASKS.TEXT_EMBEDDING;
+
+    const expected = {
+      copy_from: 'ml.inference.dest.predicted_value',
+      description: "Copy the predicted_value to 'dest'",
+      field: destinationField,
+    };
+
+    expect(getSetProcessorForInferenceType(destinationField, inferenceType)).toEqual(expected);
+  });
+
+  it('should return undefined for unknown inferenceType', () => {
+    const inferenceType = 'wrongInferenceType';
+
+    expect(getSetProcessorForInferenceType(destinationField, inferenceType)).toBeUndefined();
+  });
+});
+
+describe('generateMlInferencePipelineBody lib function', () => {
+  const expected: MlInferencePipeline = {
+    description: 'my-description',
+    processors: [
+      {
+        remove: {
+          field: 'ml.inference.my-destination-field',
+          ignore_missing: true,
+        },
+      },
+      {
+        inference: {
+          field_map: {
+            'my-source-field': 'MODEL_INPUT_FIELD',
+          },
+          model_id: 'test_id',
+          on_failure: [
+            {
+              append: {
+                field: '_source._ingest.inference_errors',
+                value: [
+                  {
+                    message:
+                      "Processor 'inference' in pipeline 'my-pipeline' failed with message '{{ _ingest.on_failure_message }}'",
+                    pipeline: 'my-pipeline',
+                    timestamp: '{{{ _ingest.timestamp }}}',
+                  },
+                ],
+              },
+            },
+          ],
+          target_field: 'ml.inference.my-destination-field',
+        },
+      },
+      {
+        append: {
+          field: '_source._ingest.processors',
+          value: [
+            {
+              model_version: '1',
+              pipeline: 'my-pipeline',
+              processed_timestamp: '{{{ _ingest.timestamp }}}',
+              types: ['pytorch', 'ner'],
+            },
+          ],
+        },
+      },
+    ],
+    version: 1,
+  };
+
+  it('should return something expected', () => {
+    const actual = generateMlInferencePipelineBody({
+      description: 'my-description',
+      destinationField: 'my-destination-field',
+      model: mockModel,
+      pipelineName: 'my-pipeline',
+      sourceField: 'my-source-field',
+    });
+
+    expect(actual.processors?.length).toEqual(3);
+    expect(actual).toEqual(expected);
+  });
+
+  it('should return something expected 2', () => {
+    const mockTextClassificationModel: MlTrainedModelConfig = {
+      ...mockModel,
+      ...{ inference_config: { text_classification: {} } },
+    };
+    const actual = generateMlInferencePipelineBody({
+      description: 'my-description',
+      destinationField: 'my-destination-field',
+      model: mockTextClassificationModel,
+      pipelineName: 'my-pipeline',
+      sourceField: 'my-source-field',
+    });
+
+    // includes the extra set processor
+    expect(actual.processors?.length).toEqual(4);
```

Comment:
I wanted to do something that expected specific objects with specific keys, etc, but I didn't find that. This is a pretty lame expectation and will be difficult to maintain because it's not obvious what it's actually checking.